cranelift/codegen/src/legalizer/table.rs

```diff
@@ -77,11 +90,29 @@ fn compute_addr(
         offset = pos.ins().imul_imm(index, element_size as i64);
     }
 
-    if element_offset == Offset32::new(0) {
-        pos.func.dfg.replace(inst).iadd(base, offset);
+    let element_addr = if element_offset == Offset32::new(0) {
+        pos.ins().iadd(base, offset)
     } else {
         let imm: i64 = element_offset.into();
         offset = pos.ins().iadd(base, offset);
-        pos.func.dfg.replace(inst).iadd_imm(offset, imm);
-    }
+        pos.ins().iadd_imm(offset, imm)
+    };
+
+    let element_addr = if let Some((index, bound)) = spectre_oob_cmp {
+        let flags = pos.ins().ifcmp(index, bound);
+        // If out-of-bounds, choose the table base on the misspeculation path.
+        pos.ins().selectif_spectre_guard(
+            addr_ty,
+            IntCC::UnsignedGreaterThanOrEqual,
+            flags,
+            base,
+            element_addr,
+        )
+    } else {
+        element_addr
+    };
+    let new_inst = pos.func.dfg.value_def(element_addr).inst().unwrap();
+
+    pos.func.dfg.replace_with_aliases(inst, new_inst);
+    pos.remove_inst();
```

Comment:
This removes new_inst and not inst, right?

Reply:
No, it removes `inst`; you can verify by looking at the added tests in this PR that the conditional move (`selectif_spectre_guard`) is present. The reason is that `replace_with_aliases` replaces uses of `inst`'s results but the cursor still remains at `inst`.